src/jit/gentree.cpp

```diff
@@ -6586,115 +6799,165 @@ GenTreePtr Compiler::gtNewAssignNode(GenTreePtr dst, GenTreePtr src)
 }
 
 // Creates a new Obj node.
-GenTreeObj* Compiler::gtNewObjNode(CORINFO_CLASS_HANDLE structHnd, GenTree* addr)
+GenTree* Compiler::gtNewObjNode(CORINFO_CLASS_HANDLE structHnd, GenTree* addr)
 {
     var_types nodeType = impNormStructType(structHnd);
     assert(varTypeIsStruct(nodeType));
-    GenTreeObj* objNode = new (this, GT_OBJ) GenTreeObj(nodeType, addr, structHnd);
+    unsigned size = info.compCompHnd->getClassSize(structHnd);
```

Comment:
I don't believe that we are. I originally had the call to `getClassGCLayout()` here, but found that it caused additional JIT-EE calls that did not exist before. I am nearly certain that I have not added any *new* JIT-EE calls (via SuperPmi), but there were already places that were calling the same JIT-EE call multiple times - I don't *think* I have increased that number, but it is more difficult to verify than that we are not making any *new* calls.

---
In reply to: [77384418](https://github.com/dotnet/coreclr/pull/7027#discussion_r77384418) [](ancestors = 77384418)